src/rust/engine/dep_inference/src/python/mod.rs

```diff
@@ -102,11 +109,46 @@ impl ImportCollector<'_> {
         code::at_range(self.code, range)
     }
 
-    fn string_at(&self, range: tree_sitter::Range) -> &str {
-        // https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
-        self.code_at(range)
-            .trim_start_matches(|c| "rRuUfFbB".contains(c))
-            .trim_matches(|c| "'\"".contains(c))
```

Comment:
Based on your comments, I think you handled the `r"..."`, `u"..."`, `f"..."`, and `b"..."` based on the nodes in a concatenated string, right? Or at least f-strings. 

What about excluding strings that have quote marks? Are you relying on the python-side post processing with `isidentifier()` to exclude that?

Reply:
Yes, the `string_start` and `string_end` nodes represent the prefix and quote marks. This PR focuses on the `string_content` nodes so the quote marks are already excluded.

We could do that exclusion in Rust code, but I don't believe the Rust code already does that. This PR is focused solely on adding support for concatenated strings. Any changes in what we accept should be a separate follow-on PR so they are easier to review.